File: loops/stars.js


const stars = num => {
    let a = [], b = [];
    for (let x = 1; x <= num; x++) {
        a.push(x);
        if (x < num) {
            b.push(x);
        }
    }
    let nums = a.concat(b.reverse());

    for (let x in nums) {
        let temp = [];
        let y = 0;
        while (y < nums[x]) {
            temp.push('*');
            y++;
        }
        console.log(temp.join(' '));
    }
}


stars(5);

File: loops/stars.py


def stars(n):
    nums = range(1, n + 1) + range(1, n)[::-1]

    for x in nums:
        temp = []
        y = 0
        while y < x:
            temp.append('*')
            y += 1
        print(" ".join(temp))


stars(5)